images/namesrv/alpine/runserver-customize.sh

```diff
@@ -54,8 +54,8 @@ calculate_heap_sizes()
 {
     case "`uname`" in
         Linux)
-            system_memory_in_mb=`free -m| sed -n '2p' | awk '{print $2}'`
-            system_cpu_cores=`egrep -c 'processor([[:space:]]+):.*' /proc/cpuinfo`
+            system_memory_in_mb=$(($(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)/1024/1024))
+            system_cpu_cores=$(($(cat /sys/fs/cgroup/cpu/cpu.cfs_quota_us)/100000))
```

Comment:
It would be better to check memory and cpu from cgroup are smaller than system memory and cpu, because resource limits may not be set.

Reply:
This is better,and I fixed it.Although i think it's not correct if anone use it in the kubernetes without memory limit.